src/lseq/mod.rs

```diff
@@ -0,0 +1,146 @@
+/// Contains the implementation of the exponential tree for LSeq
+pub mod ident;
+
+use ident::*;
+use serde::{Deserialize, Serialize};
+
+use crate::traits::CmRDT;
+
+// identifier, clock, site id, char
+type Entry = (Identifier, u64, u32, char);
+
+/// LSeq tree
+///
+/// An LSeq tree is a CRDT for storing sequences of data (Strings, ordered lists)
+/// Internally it works by viewing each character as the leaf of a giant tree.
+/// The path that leads to a given character is called the 'identifier' of that character
+///
+/// LSeq is very similar to the LOGOOT algorithm for representing sequences. The major change is
+/// that LSeq uses an **exponential** tree to store data. That means that at each level of the tree
+/// the space doubles. This helps prevent growth of identifier sizes.
+//#[derive(Serialize, Deserialize)]
+pub struct LSeq {
+    text: Vec<Entry>, // gen: NameGenerator
+    gen: IdentGen,
+    clock: u64,
+}
+
+/// Operations that can be performed on an LSeq tree
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(tag = "op")]
+pub enum Op {
+    /// Insert a character
+    Insert {
+        /// Identifier to insert at
+        #[serde(flatten)]
+        id: Identifier,
+        /// clock of site that issued insertion
+        clock: u64,
```

Comment:
the clock and site id should both be wrapped in newtypes to make them more clear.